test/integration/backward_compatible/sql/DecodeAndSerializationTest.js

```diff
@@ -0,0 +1,733 @@
+            [-0.7, 0.7]
+        );
+        const rowMetadata = await result.getRowMetadata();
+        rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type.should.be.eq(SqlColumnType.DOUBLE);
+
+        const rows = [];
+
+        for await (const row of result) {
+            rows.push(row);
+        }
+
+        const expectedKeys = [1, 2, 3, 4, 5, 6];
+        const expectedValues = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6];
+
+        rows.length.should.be.eq(expectedValues.length);
+
+        for (let i = 0; i < rows.length; i++) {
+            (rows[i]['this'] - expectedValues[i]).should.be.lessThan(1e-5);
+            rows[i]['__key'].should.be.eq(expectedKeys[i]);
+        }
+    });
+    it('should be able to decode/serialize DATE', async function () {
+        await basicSetup();
+        const script =
+            `
+            var map = instance_0.getMap("${mapName}");
+            for (var key = 1; key < 12; key++) {
+                map.set(new java.lang.Integer(key), java.time.LocalDate.of(key+2,key+1,key));
+            }
+        `;
+        await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
+        const result = client.getSqlService().execute(
+            `SELECT * FROM ${mapName} WHERE this > CAST (? AS DATE) AND this < CAST (? AS DATE) ORDER BY __key ASC`,
+            [new HzLocalDate(1, 1, 1), new HzLocalDate(5, 5, 5)]
+        );
+        const rowMetadata = await result.getRowMetadata();
+        rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type.should.be.eq(SqlColumnType.DATE);
+
+        const rows = [];
+
+        for await (const row of result) {
+            rows.push(row);
+        }
+
+        const expectedKeys = [1, 2, 3];
+        const expectedBaseValues = {
+            year: 3,
+            month: 2,
+            date: 1
+        };
+        rows.length.should.be.eq(expectedKeys.length);
+
+        for (let i = 0; i < rows.length; i++) {
+            const date = rows[i]['this'];
+            date.should.be.instanceof(HzLocalDate);
+
+            date.getYear().should.be.eq(expectedBaseValues.year + i);
+            date.getMonth().should.be.eq(expectedBaseValues.month + i);
+            date.getDate().should.be.eq(expectedBaseValues.date + i);
+
+            rows[i]['__key'].should.be.eq(expectedKeys[i]);
+        }
+    });
+    it('should be able to decode/serialize TIME', async function () {
+        await basicSetup();
+        const script = `
+                    var map = instance_0.getMap("${mapName}");
+                    for (var key = 1; key < 12; key++) {
+                        map.set(new java.lang.Integer(key), java.time.LocalTime.of(key+3,key+2,key+1,key + 1e8));
+                    }
+                `;
+        await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
+        const result = client.getSqlService().execute(
+            `SELECT * FROM ${mapName} WHERE this > CAST (? AS TIME) AND this < CAST (? AS TIME) ORDER BY __key ASC`,
```

Comment:
Can you remind me why do we need these casts?

Reply:
Because for date time related classes, strings are sent behind the scenes, which are needed to be casted.